benchmarks/kernels/benchmark_lora.py

```diff
@@ -0,0 +1,868 @@
+        if op_type == OpType.BGMV_EXPAND_SLICE:
+            return self.as_bgmv_expand_slice_kwargs()
+        raise ValueError(f"Unrecognized optype {self}")
+
+
+def bench_optype(ctx: BenchmarkContext,
+                 arg_pool_size: int,
+                 op_type: OpType,
+                 with_cuda_graph: bool = False) -> TMeasurement:
+
+    assert arg_pool_size >= 1
+
+    # BenchmarkContext -> BenchmarkTensors
+    bench_tensors : List[BenchmarkTensors] = \
+        [BenchmarkTensors.make(ctx, op_type) for _ in range(arg_pool_size)]
+    for bt in bench_tensors:
+        bt.sanity_check()
+
+    # BenchmarkTensors -> Dict (kwargs)
+    kwargs_list = [bt.bench_fn_kwargs(op_type) for bt in bench_tensors]
+
+    # Merge into a single kwargs and quality arguments as ArgPool
+    kwargs = {k: ArgPool([]) for k in kwargs_list[0]}
+    for _kwargs in kwargs_list:
+        for k, v in _kwargs.items():
+            kwargs[k].values.append(v)
+
+    description = f"{op_type.name}({bench_tensors[0].io_types()})"
+    cuda_graph_params = CudaGraphBenchParams(
+        num_ops_in_cuda_graph=arg_pool_size) if with_cuda_graph else None
+    with Bench(cuda_graph_params,
+               ctx.bench_label(), ctx.bench_sublabel(op_type), description,
+               op_type.bench_fn(), **kwargs) as bench:
+        return bench.run()
+
+
+def bench_baseline(ctx: BenchmarkContext,
+                   arg_pool_size: int,
+                   op_type: OpType,
+                   with_cuda_graph: bool = False) -> TMeasurement:
+
+    batch_size, hidden_size, lora_rank, seq_length, dtype = (ctx.batch_size,
+                                                             ctx.hidden_size,
+                                                             ctx.lora_rank,
+                                                             ctx.seq_length,
+                                                             ctx.dtype)
+
+    m, k, n = op_type.mkn(batch_size, seq_length, hidden_size, lora_rank)
+    if op_type.is_expand_slice_fn():
+        # For a fairer comparison.
+        n = n * ctx.num_slices
+
+    # Get matmul input and output tensors for A x B = C
+    As, Bs, Cs = [], [], []
+    for _ in range(arg_pool_size):
+        As.append(torch.rand((m, k), dtype=dtype).to("cuda"))
+        Bs.append(torch.rand((n, k), dtype=dtype).to("cuda").t())
+        Cs.append(torch.rand((m, n), dtype=dtype).to("cuda"))
+
+    # Make torch.mm kwargs
+    mm_kwargs = {'input': ArgPool(As), 'mat2': ArgPool(Bs), 'out': ArgPool(Cs)}
+
+    description = (f"torch.mm({dtype_to_str(dtype)}"
+                   f"x{dtype_to_str(dtype)}"
+                   f"=>{dtype_to_str(dtype)})")
+    cuda_graph_params = CudaGraphBenchParams(
+        num_ops_in_cuda_graph=arg_pool_size) if with_cuda_graph else None
+    with Bench(cuda_graph_params, ctx.bench_label(),
+               ctx.bench_sublabel(op_type), description, torch.mm,
+               **mm_kwargs) as bench:
```

Comment:
QQ:Does `torch.mm` support group gemm? If not, as baseline, how does it compute multi-lora gemm?